applications: asset_tracker_v2: update bsec to bsec2 (#9968)
nrfconnect/sdk-nrf: applications/asset_tracker_v2/doc/sensor_module.rst

If enabled, the BSEC library is used instead of the BME680 Zephyr driver to provide sensor readings from the BME680 for temperature, humidity, and atmospheric pressure.
In addition, the BSEC driver provides an additional sensor reading, indoor air quality (IAQ), which is a metric given in between 0-500 range, which estimates the air quality of the environment.

> Please note that using the BSEC library requires accepting a separate license agreement: https://github.com/boschsensortec/Bosch-BSEC2-Library.

Reviewer: ```suggestion
.. note::
   Using the BSEC library requires accepting a separate license agreement: https://github.com/boschsensortec/Bosch-BSEC2-Library.
```
Add BSEC license agreement to the links.txt and use it as guided here https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/doc_styleguide.html#external-links

``.. _`BSEC license agreement`: https://github.com/boschsensortec/Bosch-BSEC2-Library`` (this is how it should be in links.txt, under the heading github)
``  `BSEC license agreement`_`` (this is how to use the link in the doc)

Author: done

Perform the following steps to enable BSEC:

> 1. Disable the Zephyr BME680 driver by setting :kconfig:option:`CONFIG_BME680` to false.

Reviewer: ```suggestion
1. To disable the Zephyr BME680 driver, set the :kconfig:option:`CONFIG_BME680` Kconfig option to false.
```

> #. Enable the external sensors API BSEC integration layer by enabling :ref:`CONFIG_EXTERNAL_SENSORS_BME680_BSEC <CONFIG_EXTERNAL_SENSORS_BME680_BSEC>` option.

Reviewer: ```suggestion
#. To enable the external sensors API BSEC integration layer use the :ref:`CONFIG_EXTERNAL_SENSORS_BME680_BSEC <CONFIG_EXTERNAL_SENSORS_BME680_BSEC>` Kconfig option.
```

If enabled, the BSEC library is used instead of the BME680 Zephyr driver to provide sensor readings from the BME680 for temperature, humidity, and atmospheric pressure.
In addition, the BSEC driver provides an additional sensor reading, indoor air quality (IAQ), which is a metric given in between 0-500 range, which estimates the air quality of the environment.

.. note::

>    Using the BSEC library requires accepting a separate license agreement: `BSEC`_

Reviewer: ```suggestion
   Using the BSEC library requires accepting a separate license agreement.
   For details, see `BSEC`_.
```
